enhancements/cluster-logging/logs-observability-openshift-io-apis.md

```diff
@@ -0,0 +1,602 @@
+* Managing and deploying a cluster-wide log forwarder
+
+**cluster-observability-operator** is an operator responsible for:
+
+* managing and deploying observability operands (e.g. LokiStack, ClusterLogForwarder, Tracing) and console plugins (e.g console-logging-plugin)
+
+**loki-operator** is an operator responsible for managing a loki stack.
+
+**cluster-logging-operator** is an operator responsible for managing log collection and forwarding.
+
+The cluster administrator does the following:
+
+1. Deploys the Red Hat **cluster-observability-operator**
+1. Deploys the Red Hat **loki-operator**
+1. Deploys an instance of **LokiStack** in the `openshift-logging` namespace
+1. Deploys the Red Hat **cluster-logging-operator**
+1. Creates a **ClusterLogForwarder** custom resource for the **LokiStack**
+
+The **cluster-observability-operator**:
+1. Deploys the console-logging-plugin for reading logs in the OpenShift console
+
+The **loki-operator**:
+1. Deploys the **LokiStack** for storing logs on-cluster
+
+The **cluster-logging-operator**:
+
+1. Deploys the log collector to forward logs to log storage in the `openshift-logging` namespace
+
+### API Extensions
+
+This API defines the following opinionated input sources which is a continuation of prior cluster logging versions:
+
+* **application**: Logs of container workloads running in all namespaces except **default**, **openshift***, and **kube***
+* **infrastructure**: journald logs from OpenShift nodes and container workloads running only in namespaces **default**, **openshift***, and **kube***
+* **audit**: The logs from OpenShift nodes written to the node filesystem by: Kubernetes API server, OpenShift API server, Auditd, and OpenShift Virtual Network (OVN).
+
+These are **reserved** words that represent input sources that can be referenced by a pipeline without an explicit input specification.
+
+More explicit specification of **audit** and **infrastructure** logs is allowed by creating a named input of that type and specifiying at least one of the allowed sources.
+
+This is a namespaced resource that follows the rules and [design](https://github.com/openshift/enhancements/blob/master/enhancements/cluster-logging/multi-cluster-log-forwarder.md) described in the multi-ClusterLogForwarder proposal with the following exceptions:
+
+* Drops the `legacy` mode described in the proposal.
+* Moves collector specification to the **ClusterLogForwarder**
+
+#### ClusterLogForwarer CustomResourceDefinition:
+
+Following defines the next version of a ClusterLogForwarder. **Note:** The next version of this resources is part of a new API group to align log collection with
+the objectives of Red Hat observability.  
+
+```yaml
+apiVersion: "observability.openshift.io/v1"
+kind: ClusterLogForwarder
+metadata:
+  name:
+spec:
+  managementState:   #enum: Managed, Unmanaged
+  serviceAccount:
+    name:
+  collector:
+    resources:       #corev1.ResourceRequirements
```

Comment:
> In general it's advised not to re-use upstream types.

What is the recommended alternative?  Alias the types? re-implement?  There seems to be a caveat to this statement IMO.  I would think the intent of "core" types is to have slow evolution.  Also, there is always a danger of a dependency changing so I'm not certain how you get away from that.  We are fairly slow at bumping dependencies and most if not all of them are stable; we are not using anything with high churn.

> Is this passed directly through to a pod spec? If so, can probably excuse that

It is directly passed through to help with scheduling.

Reply:
So the advice normally is to implement your own type. For example, this is common primarily in the `corev1.ObjectReference` and its usage. Where people just want to have a secret name and namespace, but have a number of other unused fields as well.

Given this is a resource requirement, I expect you are just plumbing it through to a podSpec? If that's correct, then the usage here is fine. It would only be better to implement your own alternative if you are using it for some other purpose, and just used the type because it was handy that it looked like what you wanted already (like the object references)